sdk/search/Azure.Search.Documents/src/Indexes/Models/IndexingParameters.cs

```diff
@@ -1,20 +1,230 @@
+                }
+            }
+
+            /// <inheritdoc />
+            public ICollection<string> Keys { get; }
+
+            /// <inheritdoc />
+            public ICollection<object> Values { get; }
+
+            /// <inheritdoc />
+            public int Count =>
+                Configuration?.Count ?? 0;
+
+            /// <inheritdoc />
+            public bool IsReadOnly =>
+                Configuration?.IsReadOnly ?? false;
+
+            /// <summary>
+            /// Gets the <see cref="IndexingParametersConfiguration"/> from the parent <see cref="IndexingParameters"/>.
+            /// You should call <see cref="EnsureInitialized"/> first to ensure it is initialized if required.
+            /// </summary>
+            private IDictionary<string, object> Configuration =>
+                _parent._indexingParametersConfiguration?.Aggregate();
+
+            /// <inheritdoc />
+            public void Add(string key, object value)
+            {
+                EnsureInitialized();
+
+                Configuration.Add(key, value);
+            }
+
+            /// <inheritdoc />
+            public void Add(KeyValuePair<string, object> item)
+            {
+                EnsureInitialized();
+
+                Configuration.Add(item);
+            }
+
+            /// <inheritdoc />
+            public void Clear() =>
+                Configuration?.Clear();
+
+            /// <inheritdoc />
+            public bool Contains(KeyValuePair<string, object> item) =>
+                Configuration?.Contains(item) ?? false;
+
+            /// <inheritdoc />
+            public bool ContainsKey(string key) =>
+                Configuration?.ContainsKey(key) ?? false;
+
+            /// <inheritdoc />
+            public void CopyTo(KeyValuePair<string, object>[] array, int arrayIndex) =>
+                Configuration?.CopyTo(array, arrayIndex);
+
+            /// <inheritdoc />
+            public IEnumerator<KeyValuePair<string, object>> GetEnumerator() =>
+                Configuration?.GetEnumerator() ?? Enumerable.Empty<KeyValuePair<string, object>>().GetEnumerator();
+
+            /// <inheritdoc />
+            public bool Remove(string key) =>
+                Configuration?.Remove(key) ?? false;
+
+            /// <inheritdoc />
+            public bool Remove(KeyValuePair<string, object> item) =>
+                Configuration?.Remove(item) ?? false;
+
+            /// <inheritdoc />
+            public bool TryGetValue(string key, out object value)
+            {
+                if (Configuration != null)
+                {
+                    return TryGetValue(key, out value);
+                }
+
+                value = null;
+                return false;
+            }
+
+            /// <summary>
+            /// Initializes the <see cref="IndexingParametersConfiguration"/> on the parent <see cref="IndexingParameters"/> if null.
+            /// </summary>
+            private void EnsureInitialized() =>
```

Comment:
Why do you need to lazy-initialize the _indexingParametersConfiguration? I would think we should lazy initialize the configuration dictionary instead.

Reply:
Which configuration dictionary? If you mean `IndexingPArameters.Configuration`, that was an `IDictionary<string, object>` before and never null. Sure, we could lazy-init it but I don't see much point. The only reason I'm doing this here is to avoid a potential concurrency issue when initializing the now-model on the parent. The deserializer allows it to be null, which wasn't true for the old `configuration`. If I had some way to provide some hint to the serializing, that would help; however, that's not so straight forward since the `IndexingParametersConfiguration` as a dictionary is only reflecting the `AdditionalProperties` - not itself. That's what the aggregating dictionary is for.